core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -49,6 +50,7 @@
   static final int DEFAULT_TABLE_FORMAT_VERSION = 1;
   static final int SUPPORTED_TABLE_FORMAT_VERSION = 2;
   static final int INITIAL_SPEC_ID = 0;
+  static final int INITIAL_SORT_ORDER_ID = 0;
```

Comment:
If we want to reserve unsorted, all we would do is change this to 1 and add checks for passing in an unsorted order.

Reply:
Reserved.